src/editor/app.js

```diff
@@ -22,12 +27,14 @@ function isCleanOrForced(generation) {
 UiDriver.registerEventHandler("C_CMD_MARK_CLEAN", function(msg, data, prevReturn) {
     forceDirty = false;
     changeGeneration = editor.changeGeneration(true);
-    UiDriver.sendMessage("J_EVT_CLEAN_CHANGED", isCleanOrForced(changeGeneration));
+    UiDriver.proxy.clean = isCleanOrForced(changeGeneration);
+    UiDriver.proxy.sendEditorEvent("J_EVT_CLEAN_CHANGED", isCleanOrForced(changeGeneration));
```

Comment:
Its actually just there as a backwards compatibility with the old QWebKit events in Editor.  I need to spend some time actually reworking how events are handled in general.... and yes, setting the data when the event is sent would be a good call as well... I doubt a few extra signal/slots is gonna kill JsProxy anyhow.